File: CppPractice/Project1/Project1/Leetcode441.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int arrangeCoins(int n) {
        int ans = 0;
        while ((long)ans * (long)ans + ans <= 2 * (long)n) {
            ans++;
        }
        ans--;
        return ans;

    }
};

File: Pythonleetcode/Leetcode441.py
class Solution(object):
    def arrangeCoins(self, n):
        ans=int(n**0.5)
        i=int(n**0.5)
        n=n-(i*(i+1))/2
        while n>i:
            i=i+1
            n=n-i
            ans=ans+1
        return ans







print(Solution().arrangeCoins(1))
